idaes/surrogate/alamopy_new.py

```diff
@@ -0,0 +1,504 @@
+    CONFIG.declare("temp_path", ConfigValue(
+        default=TEMPPATH,
+        domain=str,
+        description="Path for directory to store temp files."))
+    CONFIG.declare("filename", ConfigValue(
+        default=None,
+        domain=str,
+        description="File name to use for temp files."))
+    CONFIG.declare("keep_files", ConfigValue(
+        default=False,
+        domain=In([True, False]),
+        description="Flag indicating whether to retain temp files."))
+
+    def __init__(self, **settings):
+        super().__init__(**settings)
+
+    def build_model(self):
+        pass
+
+    def writeAlamoInputFile(
+            self, x_reg=None, z_reg=None, x_val=None, z_val=None):
+        """Write the input file for the ALAMO executable."""
+        if x_reg is None:
+            x_reg = self._rdata_in
+        if z_reg is None:
+            z_reg = self._rdata_out
+        if x_val is None:
+            x_val = self._vdata_in
+        if z_val is None:
+            z_val = self._vdata_out
+
+        # Get number of data points to build alm file
+        if x_reg is not None:
+            n_inputs, n_rdata = x_reg.shape
+        else:
+            n_rdata = 0
+        if x_val is not None:
+            n_inputs, n_vdata = x_val.shape
+        else:
+            n_vdata = 0
+
+        if self.config.filename is not None:
+            filename = self.config.filename
+        else:
+            filename = "temp.alm"
+        almFile = os.path.join(self.config.temp_path, filename)
+        tracefile = filename.split('.')[0]+"_trace.trc"
+
+        with open(almFile, "w") as af:
+            af.write("# IDAES Alamopy input file\n")
+            af.write("NINPUTS {0}\n".format(self._n_inputs))
+            af.write("NOUTPUTS {0}\n".format(self._n_outputs))
+            af.write("XLABELS {0}\n".format(
+                " ".join(map(str, self._input_labels))))
+            af.write("ZLABELS {0}\n".format(
+                " ".join(map(str, self._output_labels))))
+            af.write("XMIN {0}\n".format(" ".join(map(str, self._input_min))))
+            af.write("XMAX {0}\n".format(" ".join(map(str, self._input_max))))
+            af.write("NDATA {0}\n".format(n_rdata))
+            af.write("NVALDATA {0}\n".format(n_vdata))
+
+            # Other options for config
+            # Can be bool, list of floats, None, Enum, float
+            # Special cases
+            if self.config.monomialpower is not None:
+                af.write("MONO {0}\n".format(len(self.config.monomialpower)))
+            if self.config.multi2power is not None:
+                af.write("MULTI2 {0}\n".format(len(self.config.multi2power)))
+            if self.config.multi3power is not None:
+                af.write("MULTI3 {0}\n".format(len(self.config.multi3power)))
+            if self.config.ratiopower is not None:
+                af.write("RATIOS {0}\n".format(len(self.config.ratiopower)))
+
+            for o in supported_options:

```

Comment:
Do you really need supported_options?  Why not iterate through self.config?

Reply:
Because the config has more options that should not be included, and seeing as it inherits from the base class we do not have full control of the config list.

Reply:
Not really a big deal, but you could probably use a sub-block for options that should be included.  Like alamo_config maybe?  Save the work of maintaining the supported list.  I don't know the details of how to use this, so if that doesn't make sense disregard.

Reply:
Yeah, I was worried about that.  I guess the right answer is whatever is most intuitive for the user.